0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+        image_df["site"] = site
+        image_df = image_df.assign(
+            Plate=[x[0] for x in image_df.site.str.split("-")],
+            Well=[x[1] for x in image_df.site.str.split("-")],
+            Site=[x[2] for x in image_df.site.str.split("-")],
+        )
+        image_list.append(image_df)
+    except FileNotFoundError:
+        print(f"{site} data not found")
+        continue
+
+image_df = pd.concat(image_list, axis="rows").reset_index(drop=True)
+print("Done concatenating image files")
+
+# add in x, y coordinates for plotting
+image_df["Site"] = image_df["Site"].astype(int)
+loc_df["Site"] = loc_df["Site"].astype(int)
+image_df = image_df.merge(loc_df, how="left", on="Site")
+
+# Plot final Cells thresholds per well
+cells_finalthresh_gg = (
+    gg.ggplot(image_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="Threshold_FinalThreshold_Cells"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_Cells_FinalThreshold_per_well.png"
+)
+cells_finalthresh_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot final Nuclei thresholds per well
+nuclei_finalthresh_gg = (
+    gg.ggplot(image_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="Threshold_FinalThreshold_Nuclei"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_Nuclei_FinalThreshold_per_well.png"
+)
+nuclei_finalthresh_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot percent Confluent regions per well
+percent_confluent_gg = (
+    gg.ggplot(image_df, gg.aes(x="x_loc", y="y_loc"))
+    + gg.geom_point(gg.aes(fill="Math_PercentConfluent"), size=10)
+    + gg.geom_text(gg.aes(label="Site"), color="lightgrey")
+    + gg.facet_wrap("~Well")
+    + gg.coord_fixed()
+    + gg.theme_bw()
+    + gg.theme(
+        axis_text=gg.element_blank(),
+        axis_title=gg.element_blank(),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+    )
+    + gg.scale_fill_cmap(name="magma")
+)
+output_file = pathlib.Path(
+    output_figuresdir, "plate_layout_PercentConfluent_per_well.png"
+)
+percent_confluent_gg.save(output_file, dpi=300, verbose=False)
+
+# Create list of sites with confluent regions
+confluent_df = image_df.loc[image_df["Math_PercentConfluent"] > 0]
```

Comment:
Another example of a potentially concerning hardcoded value - are we always going to be interested in `>0`?

Reply:
This I am confident that >0 should be hardcoded. It's asking for any sites that have confluent regions which in this case is defined by a Math_PercentConfluent >0 (as those without confluent regions are 0% confluent).

Reply:
👍